mnefun/_mnefun.py

```diff
@@ -2330,3 +2330,117 @@ def plot_reconstruction(evoked, origin=(0., 0., 0.04)):
     axs[2, 0].set_ylabel('Reconstruction')
     fig.tight_layout()
     return fig
+
+
+def chpi_snr_raw(raw, winlen, nharm, stop=None):
+    """ Compute and plot cHPI SNR from raw data.
+
+    Parameters
+    ----------
+
+    winlen : int
+        Length of window to use. Longer windows will naturally include more
+        low frequency power, resulting in lower SNR.
+
+    nharm : int
+        Number of line frequency harmonics to include in the model.
+
+    stop : int
+        If specified, stop before the end of file (seconds)
+    """
+
+    # plotting parameters
+    legend_fontsize = 12
+    legend_hspace = 30  # % of horizontal space to reserve for legend
+
+    # get some info from fiff
+    sfreq = raw.info['sfreq']
+    linefreq = raw.info['line_freq']
+    linefreqs = (np.arange(nharm + 1) + 1) * linefreq
+    buflen = int(winlen * sfreq)
+    if buflen <= 0:
+        raise ValueError('Window length should be >0')
+    cfreqs = []
+    if (len(raw.info['hpi_meas']) > 0 and
+            'coil_freq' in raw.info['hpi_meas'][0]['hpi_coils'][0]):
+        for coil in raw.info['hpi_meas'][0]['hpi_coils']:
+            cfreqs.append(coil['coil_freq'][0])
+    else:
+        raise ValueError('Cannot determine cHPI frequencies'
+                         'from raw data info')
+    print('\nNominal cHPI frequencies: ', cfreqs, ' Hz')
+    print('Sampling frequency:', sfreq, 'Hz')
+    print('Using line freqs:', linefreqs, ' Hz')
+    print('Using buffers of', buflen, 'samples =', buflen/sfreq, 'seconds\n')
+
+    raw.info['bads'] = []  # make sure to load all channels
```

Comment:
don't do this, it modifies `raw` -- if someone uses it afterward this will be bad